src/Microsoft.ML.Core/Properties/AssemblyInfo.cs

```diff
@@ -13,6 +13,7 @@
 [assembly: InternalsVisibleTo(assemblyName: "Microsoft.ML.InferenceTesting" + PublicKey.TestValue)]
 [assembly: InternalsVisibleTo(assemblyName: "Microsoft.ML.StaticPipelineTesting" + PublicKey.TestValue)]
 [assembly: InternalsVisibleTo(assemblyName: "Microsoft.ML.OnnxTransformerTest" + PublicKey.TestValue)]
+[assembly: InternalsVisibleTo(assemblyName: "Microsoft.ML.Functional.Tests" + PublicKey.TestValue)]
```

Comment:
So, the express purpose of this "functional tests" is that it does not have access to internals. This to avoid the problem of us inadvertantly adding something we *need* to internals, so we have one project where we can be sure, "these tests are using the actual power of the public API." See ref #2470, and its motivations. Maybe @rogancarr and @eerhardt can express more about this, though I think the motivations are captured adequately by the PR and its linked issue.

Reply:
YES! This should be noted to ALL devs - we must NEVER have InternalsVisibleTo the Functional.Tests. That defeats the purpose.

Note the comment in the Functional.Tests.csproj:

https://github.com/dotnet/machinelearning/blob/40abffc4eb98990c86804a05ca707a467edbe3d6/test/Microsoft.ML.Functional.Tests/Microsoft.ML.Functional.Tests.csproj#L3-L7

Maybe we need to make that comment stand out more. @yaeldekel - can you make it ALL CAPS or something that will catch people's eye, so they don't make the same change/mistake you just did? Maybe put it before and after the `SignAssembly` property.

```xml
  <PropertyGroup>
    <!-- DON'T CHANGE THIS!!! We are turning off strong naming to ensure we never add `InternalsVisibleTo` for these tests -->
    <SignAssembly>false</SignAssembly>
    <!-- DON'T CHANGE THIS!!! We are turning off strong naming to ensure we never add `InternalsVisibleTo` for these tests -->
    <PublicSign>false</PublicSign>
  </PropertyGroup>
```